index.d.ts

```diff
@@ -0,0 +1,6301 @@
+         */
+        edges(selector?: Selector): EdgeCollection;
+        /**
+         * Get elements in the graph matching the specified selector or filter function.
+         */
+        filter(
+            selector: Selector | ((ele: Singular, i: number, eles: CollectionArgument) => boolean),
+        ): CollectionReturnValue;
+
+        /**
+         * Allow for manipulation of elements without triggering multiple style calculations or multiple redraws.
+         * http://js.cytoscape.org/#cy.batch
+         *  A callback within which you can make batch updates to elements.
+         */
+        batch(callback: () => void): void;
+        /**
+         * Allow for manipulation of elements without triggering multiple style calculations or multiple redraws.
+         * http://js.cytoscape.org/#cy.batch
+         *
+         * Starts batching manually (useful for asynchronous cases).
+         */
+        startBatch(): void;
+        /**
+         * Allow for manipulation of elements without triggering multiple style calculations or multiple redraws.
+         * http://js.cytoscape.org/#cy.batch
+         *
+         * Ends batching manually (useful for asynchronous cases).
+         */
+        endBatch(): void;
+
+        /**
+         * Attaches the instance to the specified container for visualisation.
+         * http://js.cytoscape.org/#cy.mount
+         *
+         * If the core instance is headless prior to calling cy.mount(), then
+         * the instance will no longer be headless and the visualisation will
+         * be shown in the specified container. If the core instance is
+         * non-headless prior to calling cy.mount(), then the visualisation
+         * is swapped from the prior container to the specified container.
+         */
+        mount(element: Element): void;
+
+        /**
+         * Remove the instance from its current container.
+         * http://js.cytoscape.org/#cy.unmount
+         *
+         * This function sets the instance to be headless after unmounting from
+         * the current container.
+         */
+        unmount(): void;
+
+        /**
+         * A convenience function to explicitly destroy the Core.
+         * http://js.cytoscape.org/#cy.destroy
+         */
+        destroy(): void;
+
+        /**
+         * Get whether the instance of Cytoscape.js has been destroyed or not.
+         * https://js.cytoscape.org/#cy.destroyed
+         */
+        destroyed(): boolean;
+    }
+
+    /**
+     * https://js.cytoscape.org/#core/data
+     */
+    interface CoreData {
+        /**
+         * Read and write developer-defined data associated with the graph.
+         * http://js.cytoscape.org/#cy.data
+         */
+        /**
+         * Get the entire data object or a particular data field.
+         * @alias attr
+         *
+         * @param name The name of the field to get.
+         */
+        data(name?: string): any;
```

Comment:
There are declarations for ElementDataDefinition, NodeDataDefinition, EdgeDataDefinition above. Should they be used here?

Reply:
Also 'data' can't contain values that are non-serializable right? Shouldn't that be captured in the type? The 'any' type seems way too permissive. That's what scratch is for.